### ISSUE TITLE
Add dirs-only flag to lsjson

### ISSUE THREAD

**Issue by gauch** [ncw/rclone#3090]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.46

#### What problem are you are trying to solve?

Filter only directories in "lsjson" output

#### How do you think rclone should be changed to solve that?

add a flag "--dirs-only" to lsjson

**Comment by ncw** [comment]
I think we should add these two flags from lsf and unify the two implementations a bit more

```
      --dirs-only          Only list directories.
      --files-only         Only list files.
```

This would need

1. adding new options `DirsOnly` and `FilesOnly` to ListJSONOpt in [fs/operations/lsjson.go](https://github.com/ncw/rclone/blob/1c301f9f7ab8bf00d04331acc26b5195820ed1cf/fs/operations/lsjson.go#L67)
2. obeying those options in [fs/operations/lsjson.go](https://github.com/ncw/rclone/blob/1c301f9f7ab8bf00d04331acc26b5195820ed1cf/fs/operations/lsjson.go#L152)
3. flags adding in [cmd/lsjson/lsjson.go](https://github.com/ncw/rclone/blob/1c301f9f7ab8bf00d04331acc26b5195820ed1cf/cmd/lsjson/lsjson.go#L25) to set the `DirsOnly` and `FilesOnly` options
4. make the existing flags in [cmd/lsf/lsf.go](https://github.com/ncw/rclone/blob/1c301f9f7ab8bf00d04331acc26b5195820ed1cf/cmd/lsf/lsf.go#L165) use the new options in `ListJSONOpt` and [remove existing code](https://github.com/ncw/rclone/blob/1c301f9f7ab8bf00d04331acc26b5195820ed1cf/cmd/lsf/lsf.go#L198)

Extension
1. write some tests for fs/operations/lsjson.go as it doesn't have any!

That looks like a lot of work, but each step is only a few lines of code.

Do you fancy helping with this @gauch?

**Comment by calisro** [comment]
I wouldn't mind take this.  Seems simply enough and i'd like to start getting a better understanding of the code in general.

**Comment by calisro** [comment]
https://github.com/ncw/rclone/pull/3096

**Comment by ncw** [comment]
I've merged this to master now (thanks @calisro ) which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.47
